internal/lsp/file_handler.go

```diff
@@ -67,6 +67,10 @@ func (fh *fileHandler) URI() string {
 	return string(fh.uri)
 }
 
+func (fh *fileHandler) LanguageID() string {
+	return ""
```

Comment:
This makes me realize that metadata such as language ID should be tracked under file/document, not the handler. Handlers' responsibility is typically to just represent the location or pointer to the file/document - which is why e.g. version is also not exposed via handler.

It will take a bit more refactoring through the filesystem package to do this though - I'll do it in a separate commit - I hope you don't mind! 😉

Reply:
This is what I mean https://github.com/hashicorp/terraform-ls/commit/8fa3718b2ab1a00f98e1d2ce77b2b852d29f8dd6

For some reason I can't push directly to your branch, but feel free to cherry-pick that commit.

Reply:
Not a problem at all